packages/jaeger-ui/src/components/TracePage/TraceStatistics/index.tsx

```diff
@@ -392,79 +262,331 @@ export default class TraceStatistics extends Component<Props, State> {
     return allTableSpansChange;
   };
 
-  renderTableData() {
-    return this.state.tableValue.map(oneSpan => {
-      const {
-        count,
-        total,
-        avg,
-        min,
-        max,
-        selfTotal,
-        selfAvg,
-        selfMin,
-        selfMax,
-        percent,
-        color,
-        searchColor,
-        colorToPercent,
-      } = oneSpan;
-      const values: any[] = [count, total, avg, min, max, selfTotal, selfAvg, selfMin, selfMax, percent];
-      const uid = _.uniqueId('id');
-      if (!oneSpan.isDetail) {
-        return (
-          <MainTableData
-            key={uid}
-            type={oneSpan.type}
-            name={oneSpan.name}
-            searchColor={searchColor}
-            values={values}
-            columnsArray={columnsArray}
-            togglePopup={this.togglePopup}
-            valueNameSelector1={this.state.valueNameSelector1}
-            valueNameSelector2={this.state.valueNameSelector2}
-            color={color}
-            clickColumn={this.clickColumn}
-            colorToPercent={colorToPercent}
-          />
-        );
-      }
-      return (
-        <DetailTableData
-          key={uid}
-          type={oneSpan.type}
-          name={oneSpan.name}
-          searchColor={searchColor}
-          values={values}
-          columnsArray={columnsArray}
-          color={color}
-          togglePopup={this.togglePopup}
-          valueNameSelector2={this.state.valueNameSelector2}
-          colorToPercent={colorToPercent}
-        />
-      );
-    });
-  }
+  render() {
 
-  renderTableHead() {
-    const { sortAsc, sortIndex } = this.state;
-    return (
-      <tr>
-        {columnsArray.map((element: any, index: number) => (
-          <HeaderTable
-            element={element}
-            key={element.title}
-            sortIndex={sortIndex}
-            index={index}
-            sortClick={this.sortClick}
-            sortAsc={sortAsc}
-          />
-        ))}
-      </tr>
-    );
-  }
+    const onClickOption = (type:string,name:string) => {
+      if(this.state.valueNameSelector1 === 'sql.query' && type !== 'undefined')
+              this.togglePopup(name)
+    }
 
-  render() {
+    const columns: ColumnProps<ITableSpan>[] = [
+      {
+        title: 'Name',
+        dataIndex: 'name',
+        sorter: (a,b) => {
+          if(a.type === 'undefined'){
```

Comment:
* why is this checking the `.type` field but main sorting is done on `.name`?
* the sorter looks nearly identical across all columns, can it not me simplified with a helper function that takes a lambda to extract the value to sort?

Reply:
--> .type field checking is done to fix the row whose type is undefined at bottom irrespective of sorting i.e, In the previous trace statistics view "without tag: component" row is fixed at bottom irrespective of sorting see the (below picture in subgroup of frontend row)
--> Yep I simplified the sorter code and onCell code by using helper functions and committed the changes
![DraftPR](https://github.com/jaegertracing/jaeger-ui/assets/116095646/9b59ca08-1497-433b-9367-46ff1ed2a0f7)